app/controllers/cloud_volume_controller.rb

```diff
@@ -129,6 +129,22 @@ def cloud_volume_form_fields
     }
   end
 
+  def storage_manager_form_fields
+    assert_privileges("cloud_volume_new")
+    ems = find_by_id_filtered(ExtManagementSystem, params[:id])
+
+    form_fields = { :emstype => ems.emstype }
```

Comment:
So, I need to know what type the provider is. I used the approach of the `ems_common_form_controller` where the `emstype` was set.